app/jobs/application_job.rb

```diff
@@ -53,3 +105,35 @@ def zip_info_check!(metadata)
     end
   end
 end
+
+# Q: Why copy the logic out of resque-lock instead of using the gem?
+# A: It's a small amount of (MIT licensed) code, the gem is and unmaintained (last updated in 2012),
+# and the lock cleanup hook wasn't getting invoked automatically in our jobs.  Inlining the code
+# reduces indirection and dependency (on both resque-lock and on Resque itself, since we use ActiveJob
+# hooks here instead of Resque hooks).
+#
+# logic for `before_enqueue` `around_perform`/`clear_lock`, `queue_lock_key`, `before_enqueue_lock`, and
+# `lock_timeout` from resque-lock.
+# Copyright (c) Chris Wanstrath, Ray Krueger
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# Software), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# https://github.com/defunkt/resque-lock/blob/e06fc2bd26f96f4f3fe893caa49c1cd42c0c9423/LICENSE
+#
+# This additional copyright statement applies solely to the code that was taken from resque-lock.
```

Comment:
this is the first time i've re-used code in this way, i think, so definitely interested in feedback on whether i got this licensing stuff right.

Reply:
This is definitely a good faith attempt;  no idea what is correct, but I think this is sufficient.